iceoryx_examples/callbacks_in_c/ice_c_callbacks_with_context_data.c

```diff
@@ -0,0 +1,138 @@
+    const struct CounterTopic* userPayload;
+    if (iox_sub_take_chunk(subscriber, (const void**)&userPayload) == ChunkReceiveResult_SUCCESS)
+    {
+        iox_service_description_t serviceDescription = iox_sub_get_service_description(subscriber);
+        if (strcmp(serviceDescription.instanceString, "FrontLeft") == 0)
+        {
+            self->leftCache.value = *userPayload;
+            self->leftCache.isSet = true;
+        }
+        else if (strcmp(serviceDescription.instanceString, "FrontRight") == 0)
+        {
+            self->rightCache.value = *userPayload;
+            self->rightCache.isSet = true;
+        }
+        printf("received: %d\n", userPayload->counter);
+        fflush(stdout);
+    }
+
+    if (self->leftCache.isSet && self->rightCache.isSet)
+    {
+        printf("Received samples from FrontLeft and FrontRight. Sum of %d + %d = %d\n",
+               self->leftCache.value.counter,
+               self->rightCache.value.counter,
+               self->leftCache.value.counter + self->rightCache.value.counter);
+        fflush(stdout);
+        self->leftCache.isSet = false;
+        self->rightCache.isSet = false;
+    }
+}
+
+int main()
+{
+    signal(SIGINT, sigHandler);
+    signal(SIGTERM, sigHandler);
+
+    iox_runtime_init("iox-c-callback-subscriber");
+
+    // the listener starts a background thread and the callbacks of the attached events
+    // will be called in this background thread when they are triggered
+    iox_listener_storage_t listenerStorage;
+    iox_listener_t listener = iox_listener_init(&listenerStorage);
+
+    iox_sub_options_t options;
+    iox_sub_options_init(&options);
+    options.historyRequest = 10U;
+    options.queueCapacity = 5U;
+    options.nodeName = "iox-c-callback-subscriber-node";
+    iox_sub_storage_t subscriberLeftStorage, subscriberRightStorage;
+
+    CounterService counterService;
+    counterService.leftCache.isSet = false;
+    counterService.rightCache.isSet = false;
+
+    iox_sub_t subscriberLeft = iox_sub_init(&subscriberLeftStorage, "Radar", "FrontLeft", "Counter", &options);
+    iox_sub_t subscriberRight = iox_sub_init(&subscriberRightStorage, "Radar", "FrontRight", "Counter", &options);
+
+    // attach everything to the listener, from here one the callbacks are called when an event occurs
+    // we attach the pointer to counterService as context data with is then provided as second argument to
+    // the callback which allows us to modify counterService from within the callback
+    iox_listener_attach_subscriber_event_with_context_data(
+        listener, subscriberLeft, SubscriberEvent_DATA_RECEIVED, &onSampleReceivedCallback, &counterService);
+    iox_listener_attach_subscriber_event_with_context_data(
+        listener, subscriberRight, SubscriberEvent_DATA_RECEIVED, &onSampleReceivedCallback, &counterService);
+
+    // wait until someone presses CTRL+c
+    while (keepRunning)
+    {
+        sleep_for(100);
```

Comment:
Maybe a little fast.

Reply:
When I look at the CPU load with htop for this example in debug mode with enabled sanitizer it is between 0.0 and 0.1.

But for some reason chromium has a cpu load of 40% :/

Reply:
I did not mean from a CPU load point of view but for the user to follow in a meaningful way.